Sources/Plasma/Apps/plClient/plClient.cpp

```diff
@@ -373,6 +377,58 @@ bool plClient::Shutdown()
     return false;
 }
 
+void plClient::InitDLLs() {
+    hsStatusMessage("Init dlls client\n");
+
+    std::vector<plFileName> dlls = plFileSystem::ListDir("ModDLL",
+#if defined(HS_BUILD_FOR_WIN32)
+        "*.dll"
+#elif defined(HS_BUILD_FOR_APPLE)
+        "*.dylib"
+#else
+        "*.so"
+#endif
+    );
+
+    for (auto iter = dlls.begin(); iter != dlls.end(); ++iter)
+    {
+#ifdef HS_BUILD_FOR_WIN32
+        hsLibraryHndl mod = LoadLibraryW(iter->WideString().data());
+#else
+        hsLibraryHndl mod = dlopen(iter->AsString().c_str(), RTLD_LAZY | RTLD_LOCAL);
+#endif
+
+        if (mod)
+        {
+#ifdef HS_BUILD_FOR_WIN32
+            pInitGlobalsFunc initGlobals = (pInitGlobalsFunc)GetProcAddress(mod, "InitGlobals");
+#else
+            pInitGlobalsFunc initGlobals = (pInitGlobalsFunc)dlsym(mod, "InitGlobals");
+#endif
+
+            (*initGlobals)(hsgResMgr::ResMgr(), plFactory::GetTheFactory(), plgTimerCallbackMgr::Mgr(),
+                hsTimer::GetTheTimer(), plNetClientApp::GetInstance());
+            fLoadedDLLs.emplace_back(mod);
+        }
+    }
+}
+
+void plClient::ShutdownDLLs()
+{
+    for (hsLibraryHndl mod : fLoadedDLLs)
+    {
+#ifdef HS_BUILD_FOR_WIN32
+        BOOL ret = FreeLibrary(mod);
+        if (!ret)
+            hsStatusMessage("Failed to free lib\n");
+#else
+        dlclose(mod);
```

Comment:
`dlclose` also has a status return and can *theoretically* fail...

```suggestion
        int ret = dlclose(mod);
        if (ret != 0)
            hsStatusMessage(ST::format("Failed to free lib: {}", dlerror()).c_str());
```